explainer.md

```diff
@@ -0,0 +1,208 @@
+
+```js
+function checkFistGesture(inputSource, frame, renderer) {
+   for (finger of [[XRJoint.INDEX_PHALANGE_TIP, XRJoint.INDEX_METACARPAL],
+                  [XRJoint.MIDDLE_PHALANGE_TIP, XRJoint.MIDDLE_METACARPAL],
+                  [XRJoint.RING_PHALANGE_TIP, XRJoint.RING_METACARPAL],
+                  [XRJoint.PINKY_PHALANGE_TIP, XRJoint.PINKY_METACARPAL]]) {
+      let tip = finger[0];
+      let metacarpal = finger[1];
+      let tipPose = frame.getPose(inputSource.hand[tip], renderer.referenceSpace);
+      let metacarpalPose = frame.getPose(inputSource.hand[metacarpal], renderer.referenceSpace)
+      if (calculateDistance(tipPose.position, metacarpalPose.position) > minimumDistance ||
+          !checkOrientation(tipPose.orientation, metacarpalPose.orientation)) {
+         return false
+      }
+   }
+   return true;
+}
+
+function checkOrientation(tipOrientation, metacarpalOrientation) {
+   let tipDirection = applyOrientation(tipOrientation, [0, 0, 1]); // Z axis of tip
+   let palmDirection = applyOrientation(metacarpalOrientation, [0, -1, 0]) // -Y axis of metacarpal
+
+   if (1 - dotProduct(tipDirection, palmDirection) < minimumDeviation) {
+      return true;
+   } else {
+      return false;
+   }
+}
+```
+
+## Appendix: Proposed IDL
+
+```webidl
+partial interface XRInputSource {
+   XRHand? hand;
+}
+
+interface XRHand {
+    getter XRJoint(unsigned short jointIndex);
+}
+
+interface XRJoint {
+   XRSpace space;
+   float? radius;
+   
+   const unsigned short WRIST = ..;
+
+   // potentially: const unsigned short THUMB_TRAPEZIUM = ..;
+   const unsigned short THUMB_METACARPAL = ..;
+   const unsigned short THUMB_PHALANGE_PROXIMAL = ..;
+   const unsigned short THUMB_PHALANGE_DISTAL = ..;
+   const unsigned short THUMB_PHALANGE_TIP = ..;
+
+   const unsigned short INDEX_METACARPAL = ..;
+   const unsigned short INDEX_PHALANGE_PROXIMAL = ..;
+   const unsigned short INDEX_PHALANGE_INTERMEDIATE = ..;
+   const unsigned short INDEX_PHALANGE_DISTAL = ..;
+   const unsigned short INDEX_PHALANGE_TIP = ..;
+
+   const unsigned short MIDDLE_METACARPAL = ..;
+   const unsigned short MIDDLE_PHALANGE_PROXIMAL = ..;
+   const unsigned short MIDDLE_PHALANGE_INTERMEDIATE = ..;
+   const unsigned short MIDDLE_PHALANGE_DISTAL = ..;
+   const unsigned short MIDDLE_PHALANGE_TIP = ..;
+
+   const unsigned short RING_METACARPAL = ..;
+   const unsigned short RING_PHALANGE_PROXIMAL = ..;
+   const unsigned short RING_PHALANGE_INTERMEDIATE = ..;
+   const unsigned short RING_PHALANGE_DISTAL = ..;
+   const unsigned short RING_PHALANGE_TIP = ..;
+
+   const unsigned short PINKY_METACARPAL = ..;
+   const unsigned short PINKY_PHALANGE_PROXIMAL = ..;
+   const unsigned short PINKY_PHALANGE_INTERMEDIATE = ..;
+   const unsigned short PINKY_PHALANGE_DISTAL = ..;
+   const unsigned short PINKY_PHALANGE_TIP = ..;
```

Comment:
We considered using the term `PINKY` in the Windows Mixed Reality native APIs.  However, in evaluating the global acceptability of that term, we found that “pinky” is relatively unknown in non-US English locales such as the UK and Australia (though not New Zealand!), while “little finger” is still understandable in the US, especially in analogy to the “little toe”.

I'd suggest we use `LITTLE` rather than `PINKY` in these constants.

We evaluated similar concerns around `RING` - however, “ring finger” turns out to be the only non-jargon term used across all English locales for the fourth digit.

Reply:
good catch!

Reply:
For `RING`, we also did consider just using `FOURTH`.  However, we also found that there is ambiguity there in that the ring finger is both the "fourth digit" and the "third finger".

It's likely for this reason that [thesaurus.com names the "ring finger" definition](https://www.thesaurus.com/browse/ring%20finger) for which synonyms are provided the "finger next to little finger". 😅